packages/flutter_tools/lib/src/asset.dart

```diff
@@ -317,7 +319,9 @@ DevFSContent _createFontManifest(Map<String, dynamic> manifestDescriptor,
     fonts.addAll(manifestDescriptor['fonts']);
   if (fonts.isEmpty)
     return null;
-  return new DevFSStringContent(JSON.encode(fonts));
+  String json = JSON.encode(fonts);
+  json = json.replaceAll(_assetJsonLeadingParentRegExp, '"asset":"');
```

Comment:
ya, I started down that path initially with some graph traversal though I would need to handle some conversion edge cases in the behaviour that normal dart maps don't support like using NaN as map keys etc that seems to get out of hand quickly. 

I don't suppose there are easy ways in Dart to unroll mixins that are bringing in the unmodifiable behaviours?